src/controllers/StaffController.js

```diff
@@ -553,6 +625,580 @@ class StaffController {
+    const { id } = req.params;
+    if (!id) {
+      return res.status(400).json({ error: 'Missing ID' });
+    }
+    const data = req.body;
+    if (!data) {
+      return res.status(400).json({ error: 'Missing data for the request' });
+    }
+    const typeCollection = data.collection;
+    if (!typeCollection) {
+      return res.status(400).json({
+        error: 'Missing collection type for the request',
+        msg: 'Ensure the key is defined as \'collection\'',
+      });
+    }
+    const validCollectionTypes = ['Department', 'Faculty'];
+    if (!validCollectionTypes.includes(typeCollection)) {
+      return res.status(400).json({
+        error: 'Invalid collection type for the request',
+        msg: 'Only "Department" OR "Faculty" is allowed',
+      });
+    }
+    const { level } = data;
+    const adminFxn = await authClient.staffPreCheck(req);
+    if (adminFxn.error) {
+      return res.status(401).json({
+        error: adminFxn.error,
+        msg: 'Staff is not authorized to perform this action',
+      });
+    }
+    const { staff, token } = adminFxn;
+    const availableCourses = await staff.getAvailableCourses(id, typeCollection, level);
+    if (availableCourses.error) {
+      return res.status(400).json({
+        error: availableCourses.error,
+      });
+    }
+    const operatorData = await staff.getDashboardData();
+    if (!operatorData) {
+      return res.status(400).json({ error: 'Internal Server Error fetching Dashboard' });
+    }
+    return res.status(200).json({
+      operation: 'Available courses fetched successfully',
+      xToken: token,
+      availableCourses,
+      operatorData,
+    });
+  }
+
+  /**
+   * Sets available courses for the specified collection.
+   * @async
+   * @static
+   * @param {Object} req - The request object.
+   * @param {Object} res - The response object.
+   * @returns {Promise<Object>} The result of setting available courses.
+   * @throws {Error} If there is an error during the set process.
+   */
+  static async adminSetAvailableCourse(req, res) {
+    const { id } = req.params;
+    if (!id) {
+      return res.status(400).json({ error: 'Missing ID' });
+    }
+    const data = req.body;
+    if (!data) {
+      return res.status(400).json({ error: 'Missing data for the request' });
+    }
+    const typeCollection = data.collection;
+    if (!typeCollection) {
+      return res.status(400).json({
+        error: 'Missing collection type for the request',
+        msg: 'Ensure the key is defined as \'collection\'',
+      });
+    }
+    const validCollectionTypes = ['Department', 'Faculty'];
+    if (!validCollectionTypes.includes(typeCollection)) {
+      return res.status(400).json({
+        error: 'Invalid collection type for the request',
+        msg: 'Only "Department" OR "Faculty" is allowed',
+      });
+    }
+    const arrCourses = data.courses;
+    // ensure arrCourses is an array
+    if (!Array.isArray(arrCourses)) {
```

Comment:
_Same as above. Code duplication: Called method already performs this check. . ._